neo/test/rawiotest/test_neuroscoperawio.py

```diff
@@ -1,18 +1,61 @@
 import unittest
+import logging
+from pathlib import Path
 
 from neo.rawio.neuroscoperawio import NeuroScopeRawIO
 from neo.test.rawiotest.common_rawio_test import BaseTestRawIO
+from neo.test.rawiotest import rawio_compliance as compliance
+from neo.utils.datasets import get_local_testing_data_folder
 
 
-class TestNeuroScopeRawIO(BaseTestRawIO, unittest.TestCase, ):
+class TestNeuroScopeRawIO(BaseTestRawIO, unittest.TestCase):
     rawioclass = NeuroScopeRawIO
     entities_to_download = [
         'neuroscope'
     ]
     entities_to_test = [
-        'neuroscope/test1/test1'
+        'neuroscope/test1/test1',
+        'neuroscope/test1/test1.dat',
+        'neuroscope/dataset_1/YutaMouse42-151117.eeg'
     ]
 
+    def test_binary_argument_with_non_canonical_xml_file(self):
+
+        local_test_dir = get_local_testing_data_folder()
+        filename = local_test_dir / "neuroscope/test2/recording.xml"
+        binary_file = local_test_dir / "neuroscope/test2/signal1.dat"
+        reader = NeuroScopeRawIO(filename=filename, binary_file=binary_file)
+
+        txt = reader.__repr__()
+        msg = 'Before parser_header() nb_block should be NOT known'
+        assert 'nb_block' not in txt, msg
```

Comment:
Instead of checking the representation of the reader, wouldn't it be cleaner to check if the attribute `nb_block` is set, or in general no header information is present?

Reply:
```suggestion
        msg = 'Before parser_header() no header information should be present'
        assert reader.header is None, msg
```